### PR TITLE
chore: cypress selectors refactor in explore module

### PR DESCRIPTION
**Pull request by adam-stasiak**
### SUMMARY

This PR provides selectors refactor for most tests in explore test module (without visualization). I skipped some changes related to react-select components because it needs to be organized differently than normal components.
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/cypress-base/cypress/integration/explore/AdhocMetrics.test.ts [hunk 1]
```diff
@@ -70,12 +72,16 @@ describe('AdhocMetrics', () => {
...
+    cy.get('[data-test="adhoc-metric-edit-tabs"]')
+      .contains('Custom SQL')
+      .click();
+    cy.get('[data-test="metrics-edit-popover"]')
       .find('.ace_text-input')
       .type('/COUNT(DISTINCT name)', { force: true });
-    cy.get('#metrics-edit-popover').find('button').contains('Save').click();
+    cy.get('[data-test="metrics-edit-popover"]')
+      .find('[data-test="AdhocMetricEdit#save"]')
+      .contains('Save')
```
Comment (eschutho) [inline superset-frontend/cypress-base/cypress/integration/explore/AdhocMetrics.test.ts:83]: do we still need the lookups by text if we're using the data-test attributes?

File: superset-frontend/src/chart/Chart.jsx [hunk 2]
```diff
@@ -201,7 +201,10 @@ class Chart extends React.PureComponent {
         showMessage={false}
       >
         <Styles className="chart-container">
-          <div className={`slice_container ${isFaded ? ' faded' : ''}`}>
+          <div
+            className={`slice_container ${isFaded ? ' faded' : ''}`}
```
Comment (eschutho) [inline superset-frontend/src/chart/Chart.jsx:205]: this is old code, but maybe we can refactor this with `classnames` (cx)

**Review by eschutho — COMMENTED** [review]
Looks good.. overall I think if we're going to use more data-test attributes to make this less brittle, we should remove the lookups by text as well?

File: superset-frontend/cypress-base/cypress/integration/explore/AdhocFilters.test.ts [hunk 3]
```diff
@@ -77,12 +77,17 @@ describe('AdhocFilters', () => {
...
 
-    cy.get('#filter-edit-popover #adhoc-filter-edit-tabs-tab-SQL').click();
-    cy.get('#filter-edit-popover .ace_content').click();
-    cy.get('#filter-edit-popover .ace_text-input').type(
-      "'Amy' OR name = 'Bob'",
-    );
-    cy.get('#filter-edit-popover button').contains('Save').click();
+    cy.get('[data-test="filter-edit-popover"]')
+      .find('[data-test="adhoc-filter-edit-tabs"]')
+      .contains('Custom SQL')
```
Comment (rusackas) [inline superset-frontend/cypress-base/cypress/integration/explore/AdhocFilters.test.ts:82]: Curious if you think we need to do the get->find->contains->click work here, or if we should just make an explicit/specific attribute here and use it directly, e.g. `cy.get('[data-test="adhoc-filter-edit-tab-custom-sql"]').click()

File: superset-frontend/cypress-base/cypress/integration/explore/AdhocMetrics.test.ts (hunk 1 above)
Comment (rusackas) [reply superset-frontend/cypress-base/cypress/integration/explore/AdhocMetrics.test.ts:83]: Also wondering if we need the `get`->`find` chain here, or if we can just `get` the specific element we want to test on, with a specific enough attribute

File: superset-frontend/src/explore/components/QueryAndSaveBtns.jsx [hunk 4]
```diff
@@ -115,6 +115,7 @@ export default function QueryAndSaveBtns({
             data-toggle="modal"
             disabled={saveButtonDisabled}
             onClick={onSave}
+            data-test="save-button"
```
Comment (rusackas) [inline superset-frontend/src/explore/components/QueryAndSaveBtns.jsx:118]: Wondering if something more-explicit, e.g. `explore-save-button` might prevent a future issue where we have two `save-button` elements, like this one and some other Save button in a Popover or Modal.

File: superset-frontend/src/explore/components/AdhocFilterEditPopover.jsx [hunk 5]
```diff
@@ -176,6 +181,7 @@ export default class AdhocFilterEditPopover extends React.Component {
           </Tabs>
           <div>
             <Button
+              data-test="modal-save-button"
```
Comment (rusackas) [inline superset-frontend/src/explore/components/AdhocFilterEditPopover.jsx:184]: this isn't a modal, but rather a popover.

File: superset-frontend/src/chart/Chart.jsx (hunk 2 above)
Comment (rusackas) [reply superset-frontend/src/chart/Chart.jsx:205]: There are probably a million of these in the codebase... wonder if it's worth tasking someone to go on a seek and destroy mission throughout the codebase, to better establish the pattern.

File: superset-frontend/src/CRUD/CollectionTable.tsx [hunk 6]
```diff
@@ -292,7 +292,7 @@ export default class CRUDCollection extends React.PureComponent<
             </Button>
           )}
         </span>
-        <table className="table">
+        <table data-test="table" className="table">
```
Comment (rusackas) [inline superset-frontend/src/CRUD/CollectionTable.tsx:295]: maybe `crud-table` just to make sure we don't find ourselves running into an issue with two tables conflicting on a page? Sorry, I know these problems all seem hypothetical, but this is an excuse to _not_ run into the same issues we do with class names!

**Review by rusackas — CHANGES_REQUESTED** [review]
Most of this looks great! Tempted to approve, but I'm just going to mark this as needing changes, since there are enough open questions to discuss. I *think* you can request a re-review with the Github button once you've answered questions or made changes as you see fit, but just ping me if that's not the case! Thanks!

**Pull request closed without merging**